ecc_enclave/enclave/shim.h

```diff
@@ -227,6 +227,18 @@ void get_creator_name(char* msp_id,  // MSP id of organization to which transact
 // upgrade it according to what highlighted in note-3.
 extern int get_random_bytes(uint8_t* buffer, size_t length);
 
+// get_channel_id
+// --------------
+// returns the channel id string.
+// The channel id might be truncated if the buffer size is not large enough.
+void get_channel_id(char* channel_id, uint32_t max_msp_id_len, shim_ctx_ptr_t ctx);
+
+// get_msp_id
+// --------------
+// returns the msp id string.
+// The msp id might be truncated if the buffer size is not large enough.
+void get_msp_id(char* msp_id, uint32_t max_msp_id_len, shim_ctx_ptr_t ctx);
```

Comment:
what for is this msp_id? The creators? If so, why do we need a separate function to get_creator_name which also returns that?  If something else, what is it?

Reply:
it is the peer MSP. see this https://github.com/hyperledger/fabric-chaincode-go/blob/master/shim/stub.go#L138

Reply:
I have the feeling that, both, the peer MSP id and the channel id should be initially injected into the enclave when it is starting and stored in the enclave locally. That also would also mean that we don't need to implement these ocalls and give a peer a chance to inject "wrong" data for whatever reason. 
A chaincode calling get_msp_id and get_channel_id would always receive the values that are initially set during enclave creating.

As the channel id and the msp id won't change during the lifetime of the enclave, we save some lines of code and limit also the enclave interface.

Reply:
I think we had the in the previous version of the `createEnclave` see https://github.com/hyperledger-labs/fabric-private-chaincode/blob/new_design/docs/fpc-registration.puml#L105

Reply:
> what for is this msp_id?

Marcus answered for me.

> That also would also mean that we don't need to implement these ocalls and give a peer a chance to inject "wrong" data for whatever reason

Given that this is a concern, I have modified the implementation for setting these values once: either at the beginning from the Fabric shim, or from the sealed parameters.
This matches what we currently have, since the parameters are verified only later by the enclave registry, and then we keep them in sealed storage (to be implemented).

> the peer MSP id and the channel id should be initially injected into the enclave

That's a possibility, which I wanted to avoid because we should replicate these values somewhere else, instead of leveraging that Fabric already provides.

Reply:
> > That also would also mean that we don't need to implement these ocalls and give a peer a chance to inject "wrong" data for whatever reason
> 
> Given that this is a concern, I have modified the implementation for setting these values once: either at the beginning from the Fabric shim, or from the sealed parameters.
> This matches what we currently have, since the parameters are verified only later by the enclave registry, and then we keep them in sealed storage (to be implemented).

Ok, i see now where you are heading.  But in that case, please add already the CCParam data structure and initialize it already where the enclave (and its keys) are initialized, so the logic is clear how the security works.